Framework/MainDriverApi.py

```diff
@@ -947,6 +947,7 @@ def run_test_case(
             "status": sTestCaseStatus,
             "failreason": ""
         }
+        shared.Set_Shared_Variables("runid_status", "In-Progress" if sTestCaseStatus in passed_tag_list and shared.Get_Shared_Variables("runid_status")=="In-Progress" else "Blocked")
```

Comment:
the way this logic is spread out across multiple places is not good
consolidate the logic into a single function and call that function from everywhere else
+ will it always be `"In-Progress"` and `"Blocked"`? what happens when a runid is completed?

Reply:
then It will set completed